agent/intentions_endpoint.go

```diff
@@ -67,7 +67,13 @@ func (s *HTTPHandlers) IntentionCreate(resp http.ResponseWriter, req *http.Reque
 		return nil, fmt.Errorf("Failed to decode request body: %s", err)
 	}
 
-	// TODO(partitions): reject non-empty/non-default partitions from the decoded body
+	if args.Intention.DestinationPartition != "" && args.Intention.DestinationPartition != "default" {
+		return nil, BadRequestError{Reason: "Cannot specify a destination partition with this endpoint"}
+	}
+	if args.Intention.SourcePartition != "" && args.Intention.SourcePartition != "default" {
+		return nil, BadRequestError{Reason: "Cannot specify a source partition with this endpoint"}
+	}
```

Comment:
Can't think of an ENT/OSS-friendly clean way of doing this 😢

Reply:
you can use `structs.PartitionOrDefault` for this

Reply:
I tried using it like below but in OSS it fails because `PartitionOrDefault` always returns `"default"` and therefore we can't capture the error case where a user has specified a non-empty, non-default partition.

```go
// oss: args.Intention.SourcePartition = "invalidpartition"

if structs.PartitionOrDefault(args.Intention.SourcePartition) != "default" {
    return nil, BadRequestError{Reason: "Cannot specify a source partition with this endpoint"}
}

// oss: passes through (bug)
```

Reply:
It's the only endpoint I know of that don't error on partition or ns passed in oss, but because this is a legacy endpoint I think it's not worth fixing